src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -625,91 +624,100 @@ void FancyZones::ToggleEditor() noexcept
     }
 
     std::wstring editorLocation;
+    std::wstring targetMonitorLocation;
 
-    if (m_settings->GetSettings()->spanZonesAcrossMonitors)
-    {
-        std::vector<std::pair<HMONITOR, RECT>> allMonitors;
+    std::vector<std::pair<HMONITOR, RECT>> allMonitors;
+    m_dpiUnawareThread.submit(OnThreadExecutor::task_t{ [&] {
+                          allMonitors = FancyZonesUtils::GetAllMonitorRects<&MONITORINFOEX::rcWork>();
+                      } }).wait();
 
-        m_dpiUnawareThread.submit(OnThreadExecutor::task_t{ [&] {
-                              allMonitors = FancyZonesUtils::GetAllMonitorRects<&MONITORINFOEX::rcWork>();
-            } }).wait();
 
-        UINT currentDpi = 0;
-        for (const auto& monitor : allMonitors)
+    bool showDpiWarning = false;
+    for (auto& monitor : allMonitors)
+    {
+        UINT dpiX = 0;
+        UINT dpiY = 0;
+        if (GetDpiForMonitor(monitor.first, MDT_EFFECTIVE_DPI, &dpiX, &dpiY) == S_OK)
         {
-            UINT dpiX = 0;
-            UINT dpiY = 0;
-            if (GetDpiForMonitor(monitor.first, MDT_EFFECTIVE_DPI, &dpiX, &dpiY) == S_OK)
+            if (DPIAware::DEFAULT_DPI != dpiX)
             {
-                if (currentDpi == 0)
+                int width = monitor.second.right - monitor.second.left;
+                int height = monitor.second.bottom - monitor.second.top;
+                DPIAware::Convert(monitor.first, width, height);
+
+                monitor.second.right = monitor.second.left + width;
+                monitor.second.bottom = monitor.second.top + height;
+
+                if (m_settings->GetSettings()->spanZonesAcrossMonitors)
                 {
-                  currentDpi = dpiX;
-                  continue;
+                    showDpiWarning = true;
                 }
-                if (currentDpi != dpiX)
-                {
-                    MessageBoxW(NULL,
+            }
+        }
+    }
+
+    if (showDpiWarning)
+    {
+        MessageBoxW(NULL,
                     GET_RESOURCE_STRING(IDS_SPAN_ACROSS_ZONES_WARNING).c_str(),
                     GET_RESOURCE_STRING(IDS_POWERTOYS_FANCYZONES).c_str(),
                     MB_OK | MB_ICONWARNING);
-                    break;
-                }
-            }
-        }
+    }

```

Comment:
This code was executed only if `m_settings->GetSettings()->spanZonesAcrossMonitors` is true, we should keep it that way.